static/elements/chromedash-guide-edit-page.js

```diff
@@ -60,13 +60,11 @@ export class ChromedashGuideEditPage extends LitElement {
   fetchData() {
     this.loading = true;
     Promise.all([
-      window.csClient.getPermissions(),
       window.csClient.getFeature(this.featureId),
       window.csClient.getFeatureProcess(this.featureId),
       window.csClient.getFeatureProgress(this.featureId),
       window.csClient.getDismissedCues(),
-    ]).then(([user, feature, process, progress, dismissedCues]) => {
-      this.user = user;
+    ]).then(([feature, process, progress, dismissedCues]) => {
```

Comment:
Actually, let's think through one other scenario, imagine that a user visits the site, opens a second tab on the site, signs out in that tab, then absent-mindedly tries to continue using the first tab.    The user would still see themselves as signed in, until they try to submit an edit, they would get an error at that time, and their form values would be lost.  That's not a great experience.    How about we keep reloading the user permissions whenever the user navigates into an editing page?

Reply:
I like it. The guide-edit-page is like the "home page" for all the editing pages, so it's nice to re-check user permission before users enter any form pages. Also, it feels great now we have the flexibility to determine when to fetch data and when not to. 

I have made the changes. Now the user data is always refetched upon entering the edit page. I also tried the scenario you mentioned and added the following code. If the user has no permissions and somehow sneak into the edit page, a page reload would be forced. And this would triggers server-side check and redirect the user to the all features page.

```js 
// re-check permission every time users visit the edit page
const canEdit = (this.user &&
(this.user.can_edit_all || this.user.editable_features.includes(this.featureId)));
if (!canEdit) location.reload();
```